lib/Controller/CardController.php

```diff
@@ -134,6 +134,24 @@ public function unarchive($cardId) {
 		return $this->cardService->unarchive($cardId);
 	}
 
+	/**
+	 * @NoAdminRequired
+	 * @param $cardId
+	 * @return \OCP\AppFramework\Db\Entity
+	 */
+	public function done($cardId) {
```

Comment:
@stefan-niedermann Would you prefer to have those two methods also for the REST API or do you think an optional parameter on the cards PUT method is enough? 

Also we should at least increase the API version (minor) then so that the clients can detect if the new done parameter is already present.

Reply:
@desperateCoder what do you prefer?

Reply:
@juliushaertl how will react old Deck versions to the optional parameter? Will they just ignore it or throw an error?

Reply:
It would just be ignored

Reply:
Then I'd prefer the optional parameter in the PUTs.  it would be a non-breaking change and everyone is happy

Reply:
The `CardService` also has `archived` but the `CardController` doesn't should I add it or just pass down null?

Reply:
For the API only the CardApiController is relevant, so that is fine. We currently have this separation between the API that web uses and the one that is accessible for external apps. In the long term it would be good to migrate those to a common OCS API so that we don't have this duplication anymore, but that is something out of scope for this PR.